uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1034.php

```diff
@@ -7,20 +7,54 @@
 
   /**
    * Adds the category field to the Absence Type table
+   * and setup option value for absence type categories
    *
    * @return bool
    */
   public function upgrade_1034() {
+    $this->up1034_createCategoryColumn();
+    $this->up1034_createCategoryOptionValues();
+
+    return TRUE;
+  }
+
+  /**
+   * Creates the category field in Absence Type table if not existing
+   */
+  private function up1034_createCategoryColumn() {
     $absenceTypeTable = AbsenceType::getTableName();
 
     if (!SchemaHandler::checkIfFieldExists($absenceTypeTable, 'category')) {
       CRM_Core_DAO::executeQuery("
         ALTER TABLE {$absenceTypeTable}
-        ADD category varchar(10) NOT NULL 
-        COMMENT 'This is used for grouping leave types.'
+        ADD category int unsigned NOT NULL COMMENT 'This is used for grouping leave types.'
```

Comment:
Since the category field is required, it is possible that tests that were previously creating/fabricating absence types will now throw errors because of this new field. This might require modifying the fabricators to add a default value for this field.